src/bin/update_snapshots/main.rs

```diff
@@ -12,16 +12,25 @@ mod test_case;
 
 use tests::snapshot_tests;
 
+use crate::utils::{find_unused_snapshots, snapshots_path};
+
 fn main() {
+    let mut produced_snapshots = HashSet::new();
+
     println!("Updating snapshots:");
     for snapshot_test in snapshot_tests() {
-        if snapshot_test.run().is_ok() {
-            continue;
+        let was_test_successful = snapshot_test.run().is_ok();
+        if !was_test_successful {
+            println!("Test \"{}\"", snapshot_test.name);
         }
 
-        println!("Test \"{}\"", snapshot_test.name);
         for snapshot in snapshot_test.generate_snapshots().unwrap() {
```

Comment:
you are rendering everything twice?
- it's inefficient
- if we hit sth that shows non deterministically then different image will be used to detect changes and a different one will be written to disk